packages/engine/lib/provider/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+//! Contains the `Provider` trait and accompanying API, which enable trait objects to provide data
```

Comment:
Can we provide links to `Provider` and `Requisition` at least once (like `[\`request_by_type_tag\`]`? Makes consumption of the docs more ergonomic